src/main/java/com/atlassian/db/replica/internal/aurora/AuroraClusterDiscovery.java

```diff
@@ -18,8 +18,7 @@ public AuroraClusterDiscovery(AuroraConnectionDetails auroraConnectionDetails) {
     }
 
     public Collection<Database> getReplicas(Supplier<Connection> connectionSupplier) {
-        try {
-            final Connection connection = connectionSupplier.get();
+        try (final Connection connection = connectionSupplier.get()) {
```

Comment:
Perhaps it's worth adding unit-test as this is an ugly bug, WDYT?

Reply:
Ok, ` try-with-resources `
Is there option that connection will be a null ?

Reply:
I'm not sure whether connection should be closed here, especially that you just state that it's ConnectionSupplier responsibility, how about doing a fix in product?

What if the connection will try to be reused? Than it wont work with product.

Reply:
Probably whoever is using supplier should close the connection, however, the supplier can return the same connection multiple times etc. We should have more explicit contracts with highlighted ownership to solve the problem :)

Reply:
> We should have more explicit contracts with highlighted ownership to solve the problem :) 

yes, we should 🐛 
How we can measure the problem?